api/src/main.rs

```diff
@@ -1,37 +1,217 @@
+        let mut security_req = SecurityRequirement::new();
+        // Each security requirement needs to be met before access is allowed.
+        security_req.insert("User".to_owned(), Vec::new());
+        // These vvvvvvv-----^^^^^^^^ values need to match exactly!
+        Ok(RequestHeaderInput::Security(
+            "User".to_owned(),
+            security_scheme,
+            security_req,
+        ))
+    }
+}
+
+#[derive(Serialize, JsonSchema)]
+#[serde(crate = "rocket::serde")]
+struct RootResponse {
+    value: String,
+}
+
+#[openapi]
+#[get("/", format = "json")]
+async fn index(
+    pool: &State<PgPool>,
+    user: User,
+) -> Result<Json<RootResponse>, rocket::response::status::BadRequest<String>> {
     // Make a simple query to return the given parameter (use a question mark `?` instead of `$1` for MySQL)
     let test_id: i32 = match sqlx::query!("SELECT id FROM test")
         .fetch_one(pool.inner())
         .await
     {
         Ok(test_id) => test_id.id,
-        Err(e) => {
-            // One day we should handle errors better (HTTP return codes)
-            println!("Error: {}", e);
-            return "Error".to_string();
+        Err(_) => {
+            return Err(rocket::response::status::BadRequest(Some(
+                "Error getting database ID".to_string(),
+            )))
         }
     };
 
-    let output = format!("Hello {test_id}!");
+    let output = format!("Hello, {}! id: {}", user.email, test_id);
+
+    Ok(Json(RootResponse { value: output }))
+}
+
+#[derive(Serialize, JsonSchema)]
+#[serde(crate = "rocket::serde")]
+struct LoginResponse {
+    token: String,
+}
 
-    output
+#[derive(Deserialize, JsonSchema)]
+#[serde(crate = "rocket::serde")]
+struct LoginRequest {
+    email: String,
 }
 
-// #[launch]
-// async fn rocket() -> _ {
+#[allow(clippy::needless_pass_by_value)]
+#[openapi]
+#[post("/login", format = "json", data = "<login_request>")]
+fn login(
+    login_request: Json<LoginRequest>,
+    key: &State<PasetoSymmetricKey<V4, Local>>,
+) -> Result<Json<LoginResponse>, rocket::response::status::BadRequest<String>> {
+    // use a default token builder with the same PASETO version and purpose
+    let token = match PasetoBuilder::<V4, Local>::default()
+        .set_claim(SubjectClaim::from(login_request.email.as_str()))
+        .build(key)
+    {
+        Ok(token) => token,
+        Err(_) => {
+            return Err(rocket::response::status::BadRequest(Some(
+                "Error building token".to_string(),
+            )))
+        }
+    };
+
+    Ok(Json(LoginResponse { token }))
+}
 
-// }
 #[rocket::main]
 async fn main() -> Result<(), rocket::Error> {
+    #[derive(Deserialize)]
+    #[serde(crate = "rocket::serde")]
+    struct Config {
+        database_url: String,
+        paseto_secret_key: String,
+    }
+
+    let rocket = rocket::build();
+    let figment = rocket.figment();
+
+    let config: Config = figment.extract().expect("Failed to extract config.");
```

Comment:
Expect messages are, confusingly, worded for what SHOULD have happened, not the error that happened.

Think of it like _"I expect the config to be extracted, so there is no need to handle this possible panic."_

```rust
figment.extract().expect("Rocket figment should have parsed the config.");
```

https://doc.rust-lang.org/std/error/index.html#common-message-styles